dra-evolution/testdata/classes.yaml

```diff
@@ -1,162 +1,208 @@
+request:
+  requirements:
+  - device:
+      driverName: example.org-sriov
 ---
 # Request any SR-IOV NIC. This will allow the claim to be satisfied by
-# example.com or example.org NICs.
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+# example.com or example.org NICs. In contrast to the example.com-gpu above,
+# here it is assumed that "SR-IOV NIC" is a concept defined in a KEP which
+# then also standardizes an attribute that vendors can use to declare
+# that their device satisfies the requirements in that KEP. All of the
+# NIC examples use attribute names from that fictional KEP.
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: sriov-nic
-spec:
-  selector:
-    matchLabels:
-      device-type: sriov-nic
+request:
+  requirements:
+  - device:
+      selector: "'type.nic.k8s.io' in device.attributes" # Doesn't care whether it's "pf" or "vf".
 ---
 # Example.com SR-IOV NIC on a 1Gbps link
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-nic-1g
-spec:
-  driver: example.com-sriov
-  constraints: "device.bandwidth == '1G'"
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.quantityAttributes['bandwidth.nic.k8s.io'].compareTo(quantity('1G')) == 0"
 ---
 # Request an example.com 10Gbps SR-IOV NIC.
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-nic-10g
-spec:
-  driver: example.com-sriov
-  constraints: "device.bandwidth == '10G'"
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.quantityAttributes['bandwidth.nic.k8s.io'].compareTo(quantity('10G')) == 0"
 ---
 # Request an example.com 1Gbps or faster SR-IOV NIC.
-apiVersion: devmgmtproto.k8s.io/v1alpha1
-kind: DeviceClass
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
 metadata:
   name: example.com-sriov-nic-1g-or-faster
-spec:
-  driver: example.com-sriov
-  constraints: "device.bandwidth >= '1G'"
----
-#######
-# NOTE: We cannot create a class that is "any 1G NIC", because a class with a
-# selector cannot also have a claim detail, which contains the constraints.
-# Instead, we would need to create per-vendor classes with a particular label
-# (e.g., bandwidth=1G) and then create an aggregate class from those. This is
-# not great, but maybe acceptable for now.
-#
-# *Claims* can still add the Constraints; we just can't do it in the class.
-#######
-#
+request:
+  requirements:
+  - device:
+      driverName: example.com-sriov
+      selector: "device.quantityAttributes['bandwidth.nic.k8s.io'].compareTo(quantity('1G')) >= 0"
+---
+# Request any 1Gbps or faster SR-IOV NIC, regardless of the vendor.
+apiVersion: resource.k8s.io/v1alpha2
+kind: ResourceClass
+metadata:
+  name: example.com-sriov-nic-1g-or-faster
```

Comment:
Nice we can do this here. Name is wrong though :)

Reply:
Fixed.